### DISCUSSION TITLE
Installing a locally-built extension

### DISCUSSION THREAD

**Discussion by electricmonk** [gitpod-io/openvscode-server#508] [Q&A]
I'm trying to write an E2E test for an extension using Playwright and `openvscode-server`. The reason is that most of the extension's UI is a webview that cannot be tested using `vscode-test`. 

Ideally I'd package the extension, then build a Docker image with this extension, run it, and then start the test. I followed the readme instructions and came up with this Dockerfile:
```
FROM gitpod/openvscode-server:latest

ENV OPENVSCODE_SERVER_ROOT="/home/.openvscode-server"
ENV OPENVSCODE="${OPENVSCODE_SERVER_ROOT}/bin/openvscode-server"
RUN mkdir -p /tmp/exts
COPY ./my-extension.vsix ./tmp/exts/my-extension.vsix
RUN ${OPENVSCODE} --install-extension "./tmp/exts/my-extension.vsix"
```

This seems to be installing the extension as part of the image build process, but when I run the image and browse to the container, the extension doesn't seem to be installed.

Am I missing something?

**Comment by ankitr** [comment]
I wasn't able to get the README instructions for extensions to work — it seems to me that the `--install-extension` flag needs to be on the same command as the server is started. Here's what I got to work:

```dockerfile
FROM gitpod/openvscode-server:latest

ENV OPENVSCODE_SERVER_ROOT="/home/.openvscode-server"
ENV OPENVSCODE="${OPENVSCODE_SERVER_ROOT}/bin/openvscode-server"

COPY packages/vscode-extension/extension.vsix /extension.vsix

ENTRYPOINT [ "/bin/sh", "-c", "exec ${OPENVSCODE_SERVER_ROOT}/bin/openvscode-server --host 0.0.0.0 --without-connection-token \"${@}\"  --install-extension /extension.vsix --start-server", "--" ]
```
